json_test.go

```diff
@@ -5,12 +5,86 @@ package cbor_test
 
 import (
 	"encoding/json"
+	"fmt"
+	"io"
 	"reflect"
 	"testing"
 
 	"github.com/fxamacker/cbor/v2"
 )
 
+type TranscoderFunc func(io.Writer, io.Reader) error
+
+func (f TranscoderFunc) Transcode(w io.Writer, r io.Reader) error {
+	return f(w, r)
+}
+
+func ExampleTranscoder_fromJSON() {
```

Comment:
Thanks for adding these examples of transcoding!  

Maybe we can move these examples to example_transcoding_test.go.